woocommerce/woocommerce-android
# Handle network error coupon order creation

Closes: #9314 (it's been closed already, this builds up on the previous PR)


### Description


This PR adds an option to handle the coupon validation error in two different ways:

First regarding the actual coupon code validation and then by handling a network error. 

### Testing instructions


1. Go to the orders tab
2. click on + to add a new order
3. add a product
4. click on Add coupon
5. type any random thing that is not a valid coupon code in the text field
6. click on Done
7. make sure the correct error message is displayed : `We couldn't find a coupon with that code. Please try again`
8. erase the code
9. disable the internet
10. type a coupon code (valid or invalid, doesn't matter)
11. make sure the correct error message is displayed `Something went wrong when validating your coupon code. Please try again`
12. enabled internet again
13. add a valid coupon code
14. makes sure it is properly added
15. try to remove the coupon
16. make sure the code is correctly removed


### Images/gif


<img src="https://github.com/woocommerce/woocommerce-android/assets/30724184/ad1bab90-fd44-4a0c-932a-2d2919ffb006" width="350"/>
